cms/articles/tests/test_pages.py

```diff
@@ -405,3 +409,104 @@ def test_correct_template_used_for_edit_page(self):
             "wagtailadmin/panels/headline_figures/figures_used_by_ancestor_data.html",
             "Template required by Headline Figures functionality was not used",
         )
+
+    def test_related_data_route(self):
```

Comment:
I missed this when testing the original dataset implementation, but it would be good to expand the `DatasetStoryBlock` tests to check the duplicate links validation.

Specifically: https://github.com/ONSdigital/dis-wagtail/blob/main/cms/datasets/blocks.py#L53-L55 should be
```python
url_path = (
                block.value.url_path if block.block_type == "dataset_lookup" else urlparse(block.value["url"])
)
```

note the `block.value["url"].path` -> `block.value["url"]` change. The `url` value is a string

Reply:
There are already tests for this in https://github.com/ONSdigital/dis-wagtail/blob/b29adc4924c66997f40d87ae046b8f2a7900ff97/cms/datasets/tests/test_blocks.py#L19

I think it works correctly as is, the intent was to validate for duplicate URL paths, ignoring the domain, the `.path` is looked up after the call to `urlparse`, to get just the path from the string. 

I discussed this with Mebin at the time, I believe the issue with just comparing full URLs was that in testing environments we will likely be pointing the API lookup to different API environments, but users were likely to still test with URLs from our public domain. 

It probably needs a comment explaining this intent.

Reply:
Arguably https://example.com/something is different from https://ons.gov.uk/something

But let's revisit this should it come up